services/ui/clipboard/clipboard_impl.cc

```diff
@@ -27,6 +29,17 @@ class ClipboardImpl::ClipboardData {
 
   std::vector<std::string> GetMimeTypes() const {
     std::vector<std::string> types(data_types_.size());
+
+    // If there is no mime types cached, it means we need to query the system
+    // clipboard for data.
+    if (!types.size()) {
```

Comment:
if (!types.size() && delegate_) ? I think it will sound more natural.

Reply:
done